### PR TITLE
Specifying docker-build command platform for Apple Sillicon computers

### PR DESCRIPTION
**Pull request by murilommen**
## What changes are proposed in this pull request?

Adding an extra argument for building the Docker image, which already defaults to what is proposed here. It will enforce Docker to use the AMD architecture, either natively or by emulating it. Solves the problem stated on #4561 when building and image with `mlflow models build-docker` on Apple Sillicon machines. 

## How is this patch tested?

By using the first 6 steps (below) from the generated Dockerfile 

```dockerfile
FROM ubuntu:18.04
RUN apt-get -y update && apt-get install -y --no-install-recommends \
         wget \
         curl \
         nginx \
         ca-certificates \
         bzip2 \
         build-essential \
         cmake \
         openjdk-8-jdk \
         git-core \
         maven \
    && rm -rf /var/lib/apt/lists/*
# Download and setup miniconda
RUN curl -L https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh >> miniconda.sh
RUN bash ./miniconda.sh -b -p /miniconda && rm ./miniconda.sh
ENV PATH="/miniconda/bin:$PATH"
ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
ENV GUNICORN_CMD_ARGS="--timeout 60 -k gevent"
WORKDIR /opt/mlflow
RUN pip install mlflow==1.19.0
```
as an example on a Macbook M1 machine, with the command:

```
docker build -t image-name . --platform=linux/amd64 --no-cache
```

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/docker_utils.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def _build_image(image_name, entrypoint, mlflow_home=None, custom_setup_steps_ho
         _logger.info("Building docker image with name %s", image_name)
         os.system("find {cwd}/".format(cwd=cwd))
         proc = Popen(
-            ["docker", "build", "-t", image_name, "-f", "Dockerfile", "."],
+            ["docker", "build", "-t", image_name, "-f", "Dockerfile", ".", "--platform=linux/amd64"],
```
Comment (harupy) [inline mlflow/models/docker_utils.py:110]:
```suggestion
[
                "docker",
                "build",
                "-t",
                image_name,
                "-f",
                "Dockerfile",
                ".",
                # <Insert comment on why we need this>
                "--platform=linux/amd64",
            ],
```

@murilommen Thanks for the contribution! Could you insert a comment on why we need `--platform=linux/amd64`?

Comment (murilommen) [reply mlflow/models/docker_utils.py]: Done!

**Comment by harupy** [comment]
@murilommen Thanks for the update! Could you fix the DCO check failure: https://github.com/mlflow/mlflow/pull/4688/checks?check_run_id=3339688896?

File: .github/workflows/notify-dco-failure.js [hunk 2]
```diff
@@ -33,8 +33,10 @@ module.exports = async ({ context, github }) => {
   const dcoCheck = await getDcoCheck();
   if (dcoCheck.conclusion !== "success") {
     const body = [
-      `@${user.login} The DCO check failed. `,
-      `Please sign off your commits: ${dcoCheck.html_url}`,
+      `@${user.login} Thanks for the contribution! The DCO check failed. `,
+      `Please sign off your commits by following the instructions here: ${dcoCheck.html_url}. `,
+      "See https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#sign-your-work for more ",
+      "details.",
```
Comment (harupy) [inline .github/workflows/notify-dco-failure.js:39]: @murilommen Can you revert this change?

**Comment by harupy** [comment]
@murilommen Looks like something went wrong when you force pushed your commits. Could you push the fix again?

**Comment by murilommen** [comment]
@harupy been super busy these last couple of days, sorry for taking me so long on something so simple. regarding this problem with the DCO, I wasn't able to follow the steps to sign-off my commits, then I force pushed my commits... Will try to revert it and do it one more time, thanks for the patience!

File: .github/workflows/notify-dco-failure.js (hunk 2 above)
Comment (harupy) [reply .github/workflows/notify-dco-failure.js:39]: @murilommen Could you address this comment?

File: mlflow/models/docker_utils.py (hunk 1 above)
Comment (harupy) [reply mlflow/models/docker_utils.py]: Thanks!

**Review by harupy — APPROVED** [review]
LGTM!
